export GENIN="/Users/benayoun/Softwares/STAR-2.7.7a/genomes/MM10"

STAR --genomeDir $GENIN --readFilesIn ../FASTQ_Hardtrim/PeriMac_OVX_c1_A1_SHAM_RNAseq_1.fastq.gz  ../FASTQ_Hardtrim/PeriMac_OVX_c1_A1_SHAM_RNAseq_2.fastq.gz   --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OVX_c1_A1_SHAM_STAR_
STAR --genomeDir $GENIN --readFilesIn ../FASTQ_Hardtrim/PeriMac_OVX_c1_A2_OVX_RNAseq_1.fastq.gz   ../FASTQ_Hardtrim/PeriMac_OVX_c1_A2_OVX_RNAseq_2.fastq.gz    --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OVX_c1_A2_OVX_STAR_
STAR --genomeDir $GENIN --readFilesIn ../FASTQ_Hardtrim/PeriMac_OVX_c1_A3_SHAM_RNAseq_1.fastq.gz  ../FASTQ_Hardtrim/PeriMac_OVX_c1_A3_SHAM_RNAseq_2.fastq.gz   --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OVX_c1_A3_SHAM_STAR_
STAR --genomeDir $GENIN --readFilesIn ../FASTQ_Hardtrim/PeriMac_OVX_c1_A4_OVX_RNAseq_1.fastq.gz   ../FASTQ_Hardtrim/PeriMac_OVX_c1_A4_OVX_RNAseq_2.fastq.gz    --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OVX_c1_A4_OVX_STAR_
STAR --genomeDir $GENIN --readFilesIn ../FASTQ_Hardtrim/PeriMac_OVX_c1_A5_SHAM_RNAseq_1.fastq.gz  ../FASTQ_Hardtrim/PeriMac_OVX_c1_A5_SHAM_RNAseq_2.fastq.gz   --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OVX_c1_A5_SHAM_STAR_
STAR --genomeDir $GENIN --readFilesIn ../FASTQ_Hardtrim/PeriMac_OVX_c1_A6_OVX_RNAseq_1.fastq.gz   ../FASTQ_Hardtrim/PeriMac_OVX_c1_A6_OVX_RNAseq_2.fastq.gz    --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OVX_c1_A6_OVX_STAR_
STAR --genomeDir $GENIN --readFilesIn ../FASTQ_Hardtrim/PeriMac_OVX_c1_A7_SHAM_RNAseq_1.fastq.gz  ../FASTQ_Hardtrim/PeriMac_OVX_c1_A7_SHAM_RNAseq_2.fastq.gz   --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OVX_c1_A7_SHAM_STAR_
STAR --genomeDir $GENIN --readFilesIn ../FASTQ_Hardtrim/PeriMac_OVX_c1_A8_OVX_RNAseq_1.fastq.gz   ../FASTQ_Hardtrim/PeriMac_OVX_c1_A8_OVX_RNAseq_2.fastq.gz    --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OVX_c1_A8_OVX_STAR_
STAR --genomeDir $GENIN --readFilesIn ../FASTQ_Hardtrim/PeriMac_OVX_c1_A9_SHAM_RNAseq_1.fastq.gz  ../FASTQ_Hardtrim/PeriMac_OVX_c1_A9_SHAM_RNAseq_2.fastq.gz   --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OVX_c1_A9_SHAM_STAR_
STAR --genomeDir $GENIN --readFilesIn ../FASTQ_Hardtrim/PeriMac_OVX_c1_A10_OVX_RNAseq_1.fastq.gz  ../FASTQ_Hardtrim/PeriMac_OVX_c1_A10_OVX_RNAseq_2.fastq.gz   --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OVX_c1_A10_OVX_STAR_
STAR --genomeDir $GENIN --readFilesIn ../FASTQ_Hardtrim/PeriMac_OVX_c1_A12_OVX_RNAseq_1.fastq.gz  ../FASTQ_Hardtrim/PeriMac_OVX_c1_A12_OVX_RNAseq_2.fastq.gz   --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OVX_c1_A12_OVX_STAR_