# rerum counting for set 1 (Novogene mRNAseq is unstranded)

export GENIN="/Users/benayoun/Softwares/STAR-2.7.7a/genomes/MM10"

# ######################################### 
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_OF_CTL_1_RNAseq_Trimmed_1.fastq.gz ../FASTQ_HardTrim/PeriMac_OF_CTL_1_RNAseq_Trimmed_2.fastq.gz  --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OF_CTL_1_RNAseq_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_OF_CTL_2_RNAseq_Trimmed_1.fastq.gz ../FASTQ_HardTrim/PeriMac_OF_CTL_2_RNAseq_Trimmed_2.fastq.gz  --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OF_CTL_2_RNAseq_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_OF_CTL_3_RNAseq_Trimmed_1.fastq.gz ../FASTQ_HardTrim/PeriMac_OF_CTL_3_RNAseq_Trimmed_2.fastq.gz  --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OF_CTL_3_RNAseq_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_OF_CTL_4_RNAseq_Trimmed_1.fastq.gz ../FASTQ_HardTrim/PeriMac_OF_CTL_4_RNAseq_Trimmed_2.fastq.gz  --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OF_CTL_4_RNAseq_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_OF_CTL_5_RNAseq_Trimmed_1.fastq.gz ../FASTQ_HardTrim/PeriMac_OF_CTL_5_RNAseq_Trimmed_2.fastq.gz  --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OF_CTL_5_RNAseq_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_OF_E2_1_RNAseq_Trimmed_1.fastq.gz ../FASTQ_HardTrim/PeriMac_OF_E2_1_RNAseq_Trimmed_2.fastq.gz    --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OF_E2_1_RNAseq_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_OF_E2_2_RNAseq_Trimmed_1.fastq.gz ../FASTQ_HardTrim/PeriMac_OF_E2_2_RNAseq_Trimmed_2.fastq.gz    --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OF_E2_2_RNAseq_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_OF_E2_3_RNAseq_Trimmed_1.fastq.gz ../FASTQ_HardTrim/PeriMac_OF_E2_3_RNAseq_Trimmed_2.fastq.gz    --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OF_E2_3_RNAseq_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_OF_E2_4_RNAseq_Trimmed_1.fastq.gz ../FASTQ_HardTrim/PeriMac_OF_E2_4_RNAseq_Trimmed_2.fastq.gz    --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OF_E2_4_RNAseq_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_OF_E2_5_RNAseq_Trimmed_1.fastq.gz ../FASTQ_HardTrim/PeriMac_OF_E2_5_RNAseq_Trimmed_2.fastq.gz    --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_OF_E2_5_RNAseq_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_CTL_1_RNAseq_Trimmed_1.fastq.gz ../FASTQ_HardTrim/PeriMac_YF_CTL_1_RNAseq_Trimmed_2.fastq.gz  --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_CTL_1_RNAseq_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_CTL_2_RNAseq_Trimmed_1.fastq.gz ../FASTQ_HardTrim/PeriMac_YF_CTL_2_RNAseq_Trimmed_2.fastq.gz  --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_CTL_2_RNAseq_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_CTL_3_RNAseq_Trimmed_1.fastq.gz ../FASTQ_HardTrim/PeriMac_YF_CTL_3_RNAseq_Trimmed_2.fastq.gz  --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_CTL_3_RNAseq_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_CTL_4_RNAseq_Trimmed_1.fastq.gz ../FASTQ_HardTrim/PeriMac_YF_CTL_4_RNAseq_Trimmed_2.fastq.gz  --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_CTL_4_RNAseq_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_CTL_5_RNAseq_Trimmed_1.fastq.gz ../FASTQ_HardTrim/PeriMac_YF_CTL_5_RNAseq_Trimmed_2.fastq.gz  --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_CTL_5_RNAseq_STAR_    

######################################### 
featureCounts -t exon -D 1500 -p --primary -T 3 -s 1 -a /Users/benayoun/Softwares/Genomes/mm10_genes.gtf -o 2023-10-20_PeriMac_aging_E2_supplementation_experiment_CLEAN_counts.txt \
	PeriMac_YF_CTL_1_RNAseq_STAR_Aligned.sortedByCoord.out.bam \
	PeriMac_YF_CTL_2_RNAseq_STAR_Aligned.sortedByCoord.out.bam \
	PeriMac_YF_CTL_3_RNAseq_STAR_Aligned.sortedByCoord.out.bam \
	PeriMac_YF_CTL_4_RNAseq_STAR_Aligned.sortedByCoord.out.bam \
	PeriMac_YF_CTL_5_RNAseq_STAR_Aligned.sortedByCoord.out.bam \
	PeriMac_OF_CTL_1_RNAseq_STAR_Aligned.sortedByCoord.out.bam \
	PeriMac_OF_CTL_2_RNAseq_STAR_Aligned.sortedByCoord.out.bam \
	PeriMac_OF_CTL_3_RNAseq_STAR_Aligned.sortedByCoord.out.bam \
	PeriMac_OF_CTL_4_RNAseq_STAR_Aligned.sortedByCoord.out.bam \
	PeriMac_OF_CTL_5_RNAseq_STAR_Aligned.sortedByCoord.out.bam \
	PeriMac_OF_E2_1_RNAseq_STAR_Aligned.sortedByCoord.out.bam  \
	PeriMac_OF_E2_2_RNAseq_STAR_Aligned.sortedByCoord.out.bam  \
	PeriMac_OF_E2_3_RNAseq_STAR_Aligned.sortedByCoord.out.bam  \
	PeriMac_OF_E2_4_RNAseq_STAR_Aligned.sortedByCoord.out.bam  \
	PeriMac_OF_E2_5_RNAseq_STAR_Aligned.sortedByCoord.out.bam
