export GENIN="/Users/benayoun/Softwares/STAR-2.7.7a/genomes/MM10"

# ######################################### 

# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_Esr1_KO_RNAseq_rep1_2_1.fastq.gz   ../FASTQ_HardTrim/PeriMac_YF_Esr1_KO_RNAseq_rep1_2_2.fastq.gz   --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_Esr1_KO_RNAseq_rep1_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_Esr1_KO_RNAseq_rep2_4_1.fastq.gz   ../FASTQ_HardTrim/PeriMac_YF_Esr1_KO_RNAseq_rep2_4_2.fastq.gz   --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_Esr1_KO_RNAseq_rep2_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_Esr1_KO_RNAseq_rep3_6_1.fastq.gz   ../FASTQ_HardTrim/PeriMac_YF_Esr1_KO_RNAseq_rep3_6_2.fastq.gz   --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_Esr1_KO_RNAseq_rep3_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_Esr1_KO_RNAseq_rep4_8_1.fastq.gz   ../FASTQ_HardTrim/PeriMac_YF_Esr1_KO_RNAseq_rep4_8_2.fastq.gz   --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_Esr1_KO_RNAseq_rep4_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_Esr1_KO_RNAseq_rep5_10_1.fastq.gz  ../FASTQ_HardTrim/PeriMac_YF_Esr1_KO_RNAseq_rep5_10_2.fastq.gz  --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_Esr1_KO_RNAseq_rep5_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_Esr1_KO_RNAseq_rep6_12_1.fastq.gz  ../FASTQ_HardTrim/PeriMac_YF_Esr1_KO_RNAseq_rep6_12_2.fastq.gz  --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_Esr1_KO_RNAseq_rep6_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_rep1_1_1.fastq.gz   ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_rep1_1_2.fastq.gz   --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_Esr1_WT_RNAseq_rep1_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_rep2_3_1.fastq.gz   ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_rep2_3_2.fastq.gz   --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_Esr1_WT_RNAseq_rep2_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_rep3_5_1.fastq.gz   ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_rep3_5_2.fastq.gz   --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_Esr1_WT_RNAseq_rep3_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_rep4_7_1.fastq.gz   ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_rep4_7_2.fastq.gz   --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_Esr1_WT_RNAseq_rep4_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_rep5_9_1.fastq.gz   ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_rep5_9_2.fastq.gz   --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_Esr1_WT_RNAseq_rep5_STAR_    
# STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_rep6_11_1.fastq.gz  ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_rep6_11_2.fastq.gz  --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_Esr1_WT_RNAseq_rep6_STAR_    

#STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_Esr1_KO_RNAseq_DEEPER_rep1_2_1.fastq.gz ../FASTQ_HardTrim/PeriMac_YF_Esr1_KO_RNAseq_DEEPER_rep1_2_2.fastq.gz   --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_Esr1_KO_RNAseq_DEEPER_rep1_STAR_    
#STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_Esr1_KO_RNAseq_DEEPER_rep2_4_1.fastq.gz ../FASTQ_HardTrim/PeriMac_YF_Esr1_KO_RNAseq_DEEPER_rep2_4_2.fastq.gz   --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_Esr1_KO_RNAseq_DEEPER_rep2_STAR_    
#STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_Esr1_KO_RNAseq_DEEPER_rep3_6_1.fastq.gz ../FASTQ_HardTrim/PeriMac_YF_Esr1_KO_RNAseq_DEEPER_rep3_6_2.fastq.gz   --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_Esr1_KO_RNAseq_DEEPER_rep3_STAR_    
#STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_DEEPER_rep1_1_1.fastq.gz ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_DEEPER_rep1_1_2.fastq.gz   --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_Esr1_WT_RNAseq_DEEPER_rep1_STAR_    
#STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_DEEPER_rep2_3_1.fastq.gz ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_DEEPER_rep2_3_2.fastq.gz   --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_Esr1_WT_RNAseq_DEEPER_rep2_STAR_    
#STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_DEEPER_rep4_7_1.fastq.gz ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_DEEPER_rep4_7_2.fastq.gz   --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_Esr1_WT_RNAseq_DEEPER_rep4_STAR_    
#STAR --genomeDir $GENIN --readFilesIn ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_DEEPER_rep5_9_1.fastq.gz ../FASTQ_HardTrim/PeriMac_YF_Esr1_WT_RNAseq_DEEPER_rep5_9_2.fastq.gz   --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --readFilesCommand gzcat --runThreadN 1 --outFilterMultimapNmax 50 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outSAMtype BAM SortedByCoordinate  --alignEndsProtrude 15 ConcordantPair --outFileNamePrefix PeriMac_YF_Esr1_WT_RNAseq_DEEPER_rep5_STAR_    


samtools merge PeriMac_YF_Esr1_KO_RNAseq_rep1_STAR_Aligned.MERGEDLANES.bam PeriMac_YF_Esr1_KO_RNAseq_rep1_STAR_Aligned.sortedByCoord.out.bam PeriMac_YF_Esr1_KO_RNAseq_DEEPER_rep1_STAR_Aligned.sortedByCoord.out.bam
samtools merge PeriMac_YF_Esr1_KO_RNAseq_rep2_STAR_Aligned.MERGEDLANES.bam PeriMac_YF_Esr1_KO_RNAseq_rep2_STAR_Aligned.sortedByCoord.out.bam PeriMac_YF_Esr1_KO_RNAseq_DEEPER_rep2_STAR_Aligned.sortedByCoord.out.bam
samtools merge PeriMac_YF_Esr1_KO_RNAseq_rep3_STAR_Aligned.MERGEDLANES.bam PeriMac_YF_Esr1_KO_RNAseq_rep3_STAR_Aligned.sortedByCoord.out.bam PeriMac_YF_Esr1_KO_RNAseq_DEEPER_rep3_STAR_Aligned.sortedByCoord.out.bam
samtools merge PeriMac_YF_Esr1_WT_RNAseq_rep1_STAR_Aligned.MERGEDLANES.bam PeriMac_YF_Esr1_WT_RNAseq_rep1_STAR_Aligned.sortedByCoord.out.bam PeriMac_YF_Esr1_WT_RNAseq_DEEPER_rep1_STAR_Aligned.sortedByCoord.out.bam
samtools merge PeriMac_YF_Esr1_WT_RNAseq_rep2_STAR_Aligned.MERGEDLANES.bam PeriMac_YF_Esr1_WT_RNAseq_rep2_STAR_Aligned.sortedByCoord.out.bam PeriMac_YF_Esr1_WT_RNAseq_DEEPER_rep2_STAR_Aligned.sortedByCoord.out.bam
samtools merge PeriMac_YF_Esr1_WT_RNAseq_rep4_STAR_Aligned.MERGEDLANES.bam PeriMac_YF_Esr1_WT_RNAseq_rep4_STAR_Aligned.sortedByCoord.out.bam PeriMac_YF_Esr1_WT_RNAseq_DEEPER_rep4_STAR_Aligned.sortedByCoord.out.bam
samtools merge PeriMac_YF_Esr1_WT_RNAseq_rep5_STAR_Aligned.MERGEDLANES.bam PeriMac_YF_Esr1_WT_RNAseq_rep5_STAR_Aligned.sortedByCoord.out.bam PeriMac_YF_Esr1_WT_RNAseq_DEEPER_rep5_STAR_Aligned.sortedByCoord.out.bam

######################################### 
featureCounts -t exon -D 1500 -p --primary -T 3 -s 1 -a /Users/benayoun/Softwares/Genomes/mm10_genes.gtf -o 2024-10-1_PeriMac_Esr1KO_deeper_CLEAN_counts.txt \
	PeriMac_YF_Esr1_KO_RNAseq_rep1_STAR_Aligned.MERGEDLANES.bam \
	PeriMac_YF_Esr1_KO_RNAseq_rep2_STAR_Aligned.MERGEDLANES.bam \
	PeriMac_YF_Esr1_KO_RNAseq_rep3_STAR_Aligned.MERGEDLANES.bam \
	PeriMac_YF_Esr1_KO_RNAseq_rep4_STAR_Aligned.sortedByCoord.out.bam \
	PeriMac_YF_Esr1_KO_RNAseq_rep5_STAR_Aligned.sortedByCoord.out.bam \
	PeriMac_YF_Esr1_KO_RNAseq_rep6_STAR_Aligned.sortedByCoord.out.bam \
	PeriMac_YF_Esr1_WT_RNAseq_rep1_STAR_Aligned.MERGEDLANES.bam \
	PeriMac_YF_Esr1_WT_RNAseq_rep2_STAR_Aligned.MERGEDLANES.bam \
	PeriMac_YF_Esr1_WT_RNAseq_rep3_STAR_Aligned.sortedByCoord.out.bam \
	PeriMac_YF_Esr1_WT_RNAseq_rep4_STAR_Aligned.MERGEDLANES.bam \
	PeriMac_YF_Esr1_WT_RNAseq_rep5_STAR_Aligned.MERGEDLANES.bam	 \
	PeriMac_YF_Esr1_WT_RNAseq_rep6_STAR_Aligned.sortedByCoord.out.bam
